src/coreComponents/physicsSolvers/contact/ContactSolverBase.cpp

```diff
@@ -0,0 +1,300 @@
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/*
+ * ContactSolverBase.cpp
+ */
+
+#include "ContactSolverBase.hpp"
+
+#include "common/TimingMacros.hpp"
+#include "constitutive/ConstitutiveManager.hpp"
+#include "constitutive/contact/ContactSelector.hpp"
+#include "constitutive/solid/ElasticIsotropic.hpp"
+#include "finiteElement/elementFormulations/FiniteElementBase.hpp"
+#include "linearAlgebra/utilities/LAIHelperFunctions.hpp"
+#include "mesh/DomainPartition.hpp"
+#include "fieldSpecification/FieldSpecificationManager.hpp"
+#include "mesh/NodeManager.hpp"
+#include "mesh/SurfaceElementRegion.hpp"
+#include "physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.hpp"
+#include "common/GEOS_RAJA_Interface.hpp"
+
+namespace geosx
+{
+
+using namespace dataRepository;
+using namespace constitutive;
+
+ContactSolverBase::ContactSolverBase( const string & name,
+                                      Group * const parent ):
+  SolverBase( name, parent ),
+  m_solidSolverName(),
+  m_fractureRegionName(),
+  m_solidSolver( nullptr )
+{
+  registerWrapper( viewKeyStruct::solidSolverNameString(), &m_solidSolverName ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setDescription( "Name of the solid mechanics solver in the rock matrix" );
+
+  registerWrapper( viewKeyStruct::fractureRegionNameString(), &m_fractureRegionName ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setDescription( "Name of the fracture region." );
+
+  registerWrapper( viewKeyStruct::contactRelationNameString(), &m_contactRelationName ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setDescription( "Name of contact relation to enforce constraints on fracture boundary." );
+
+  this->getWrapper< string >( viewKeyStruct::discretizationString() ).
+    setInputFlag( InputFlags::FALSE );
+}
+
+ContactSolverBase::~ContactSolverBase()
+{
+  // TODO Auto-generated destructor stub
+}
+
+void ContactSolverBase::postProcessInput()
+{
+  m_solidSolver = &this->getParent().getGroup< SolidMechanicsLagrangianFEM >( m_solidSolverName );
+  SolverBase::postProcessInput();
+}
+
+void ContactSolverBase::registerDataOnMesh( dataRepository::Group & meshBodies )
+{
+  meshBodies.forSubGroups< MeshBody >( [&] ( MeshBody & meshBody )
+  {
+    MeshLevel & meshLevel = meshBody.getMeshLevel( 0 );
+
+    ElementRegionManager & elemManager = meshLevel.getElemManager();
+    {
+      elemManager.forElementRegions< SurfaceElementRegion >( [&] ( SurfaceElementRegion & region )
+      {
+        region.forElementSubRegions< EmbeddedSurfaceSubRegion >( [&]( EmbeddedSurfaceSubRegion & subRegion )
+        {
+
+          subRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::dispJumpString() ).
+            setPlotLevel( PlotLevel::LEVEL_0 ).
+            reference().resizeDimension< 1 >( 3 );
+
+          subRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::deltaDispJumpString() ).
```

Comment:
Why do we have `deltaDispJump` but not `deltaTraction`?
Shouldn't either or both make more sense?
I mean, I don't see why to distinguish. Do you have a specific reason?

Reply:
I think it's only because the `deltaDispJump`  was used by the `SolidMechanicsEmbeddedFracturesSolver` trying to be consistent with what's done for primary variables in the flow solvers with `deltaPressure`, etc. We could add `deltaTraction` as well. I think that there is still some work to do to "consistify" the way variables are updated across the solvers though.